examples/specialty_plots/skewt.py

```diff
@@ -24,66 +26,24 @@
 # The sole purpose of this class is to look at the upper, lower, or total
 # interval as appropriate and see what parts of the tick to draw, if any.
 class SkewXTick(maxis.XTick):
-    def update_position(self, loc):
-        # This ensures that the new value of the location is set before
-        # any other updates take place
-        self._loc = loc
-        super().update_position(loc)
-
-    def _has_default_loc(self):
-        return self.get_loc() is None
-
-    def _need_lower(self):
-        return (self._has_default_loc() or
-                transforms.interval_contains(self.axes.lower_xlim,
-                                             self.get_loc()))
-
-    def _need_upper(self):
-        return (self._has_default_loc() or
-                transforms.interval_contains(self.axes.upper_xlim,
-                                             self.get_loc()))
-
-    @property
-    def gridOn(self):
-        return (self._gridOn and (self._has_default_loc() or
-                transforms.interval_contains(self.get_view_interval(),
-                                             self.get_loc())))
-
-    @gridOn.setter
-    def gridOn(self, value):
-        self._gridOn = value
-
-    @property
-    def tick1On(self):
-        return self._tick1On and self._need_lower()
-
-    @tick1On.setter
-    def tick1On(self, value):
-        self._tick1On = value
-
-    @property
-    def label1On(self):
-        return self._label1On and self._need_lower()
-
-    @label1On.setter
-    def label1On(self, value):
-        self._label1On = value
-
-    @property
-    def tick2On(self):
-        return self._tick2On and self._need_upper()
-
-    @tick2On.setter
-    def tick2On(self, value):
-        self._tick2On = value
-
-    @property
-    def label2On(self):
-        return self._label2On and self._need_upper()
-
-    @label2On.setter
-    def label2On(self, value):
-        self._label2On = value
+    def draw(self, renderer):
+        with ExitStack() as stack:
```

Comment:
This is kind of mysterious for an example w/o a comment why you need it.

Reply:
added an explanation, I hope it helps